odoo/modules/module.py

```diff
@@ -311,74 +341,72 @@ def get_module_root(path):
-            non_dependencies = info['auto_install'].difference(info['depends'])
-            assert not non_dependencies,\
-                "auto_install triggers must be dependencies, found " \
-                "non-dependencies [%s] for module %s" % (
-                    ', '.join(non_dependencies), module
-                )
-        elif info['auto_install']:
-            info['auto_install'] = set(info['depends'])
-
-        info['version'] = adapt_version(info['version'])
-        return info
-
-    _logger.debug('module %s: no manifest file found %s', module, MANIFEST_NAMES)
-    return {}
+
+    if not manifest_file:
+        _logger.debug('module %s: no manifest file found %s', module, MANIFEST_NAMES)
+        return {}
+
+    manifest = copy.deepcopy(_DEFAULT_MANIFEST)
+    manifest['icon'] = get_module_icon(module)
+
+    with tools.file_open(manifest_file, mode='r') as f:
+        manifest.update(ast.literal_eval(f.read()))
+
+    if not manifest.get('description'):
+        readme_path = [opj(mod_path, x) for x in README
+                       if os.path.isfile(opj(mod_path, x))]
+        if readme_path:
+            with tools.file_open(readme_path[0]) as fd:
+                manifest['description'] = fd.read()
+
+    if not manifest.get('license'):
+        manifest['license'] = 'LGPL-3'
+        _logger.warning("Missing `license` key in manifest for %r, defaulting to LGPL-3", module)
+
+    # auto_install is either `False` (by default) in which case the module
+    # is opt-in, either a list of dependencies in which case the module is
+    # automatically installed if all dependencies are (special case: [] to
+    # always install the module), either `True` to auto-install the module
+    # in case all dependencies declared in `depends` are installed.
+    if isinstance(manifest['auto_install'], collections.abc.Iterable):
+        manifest['auto_install'] = set(manifest['auto_install'])
+        non_dependencies = manifest['auto_install'].difference(manifest['depends'])
+        assert not non_dependencies,\
+            "auto_install triggers must be dependencies, found " \
+            "non-dependencies [%s] for module %s" % (
+                ', '.join(non_dependencies), module
+            )
+    elif manifest['auto_install']:
+        manifest['auto_install'] = set(manifest['depends'])
+
+    manifest['version'] = adapt_version(manifest['version'])
+    manifest['addons_path'] = realpath(opj(mod_path, os.pardir))
+
+    return manifest
+
+@functools.lru_cache(maxsize=None)
+def load_manifest(module, mod_path=None):
+    """
+    Get the manifest dictionary from the file-system.
+
+    :param str module: The name of the module (sale, purchase, ...).
+    :param Optional[str] mod_path: The optional path to the module on
+        the file-system. It is determined by scanning the addons-paths
+        when the argument is not set.
+    :return dict: The module manifest as a dict or an empty dict
```

Comment:
```suggestion
    :return: The module manifest as a dict or an empty dict
```
`:return:` does not take any parameter, use `:rtype:` if you want to specify a programmatic type. Also prefer return*s* for grammatical readability (that is [what sphinx produces either way](https://www.odoo.com/documentation/15.0/developer/reference/backend/orm.html#odoo.fields.Date.add))

Reply:
Pfff that sucks.... nice catch thank you :+1: